src/info/mod.rs

```diff
@@ -159,29 +159,59 @@ impl Info {
     pub fn new(config: Config) -> Result<Self> {
         let git_version = cli::get_git_version();
         let repo = Repository::discover(&config.repo_path)?;
-        let internal_repo = Repo::new(&repo, config.no_merges, &config.bot_regex_pattern)?;
+        let workdir = repo.workdir().expect("non-bare repo").to_owned();
+
+        let pending_changes = std::thread::spawn({
```

Comment:
Why make this function run in  // ? Is it slow?
If not, I would rather have it encapsulated inside the `Repo` Struct.

Reply:
That's a great point! Generally, obtaining the status of a work tree performs worse the more files there are in a repository.

Doing a `git status` on the linux kernel tree for example takes git itself ~0.5s, and doing it with `git2` seems to take ~2s (while it's running in parallel with tokei, so there is some competition).

```
➜  linux git:(e1bcd7e31a6a) ✗ time /Users/byron/dev/github.com/o2sh/onefetch/target/release/onefetch
[src/info/repo.rs:445] start.elapsed() = 1.007116083s
[..]
/Users/byron/dev/github.com/o2sh/onefetch/target/release/onefetch  11.87s user 2.17s system 180% cpu 7.786 total
```

With bigger repos, I think it definitely makes sense to run it in parallel so its cost don't slow down the overall operation, this one definitely seems to be worth it.

For fun I didn't run it in parallel and got this result:

```
➜  linux git:(e1bcd7e31a6a) ✗ time /Users/byron/dev/github.com/o2sh/onefetch/target/release/onefetch
[src/info/repo.rs:445] start.elapsed() = 977.535291ms
[..]
/Users/byron/dev/github.com/o2sh/onefetch/target/release/onefetch  11.85s user 1.93s system 157% cpu 8.737 total
```